File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int cache[26] = {0};
        if (s.length() != t.length()) {
            return false;
        }
        for (char c: s) {
            cache[c - 'a']++;
        }
        for (char b: t) {
            cache[b- 'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if (cache[i] != 0) {
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> cache = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int sub = target - nums[i];
            if (cache.containsKey(sub)) {
                return new int[]{cache.get(sub), i};
            } else {
                cache.put(nums[i], i);
            }
        }
        return new int[]{};
    }
}
